src/test/java/racing/domain/car/CarsTest.java

```diff
@@ -0,0 +1,87 @@
+package racing.domain.car;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class CarsTest {
+
+    private static Stream<Arguments> Car들_중_최대_위치_값_확인() {
+        return Stream.of(
+            Arguments.of(Arrays.asList(new Car("a", 2), new Car("b", 4), new Car("C", 1))));
+    }
+
+    private static Stream<Arguments> 중복된_이름_확인() {
+        return Stream.of(
+            Arguments.of(Arrays.asList(new Car("a", 2), new Car("a", 4), new Car("C", 1))));
+    }
+
+    private static Stream<Arguments> 자동차_개수_확인() {
+        return Stream.of(Arguments.of(Arrays.asList()));
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    void Car들_중_최대_위치_값_확인(List<Car> parameterCars) {
+        final Cars cars = new Cars(parameterCars);
+        assertThat(cars.getMaxPosition()).isEqualTo(4);
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    void 중복된_이름_확인(List<Car> parameterCars) {
+        assertThatExceptionOfType(IllegalArgumentException.class).isThrownBy(
+            () -> new Cars(parameterCars)).withMessage("[ERROR] 중복된 이름은 들어갈 수 없습니다.");
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    void 자동차_개수_확인(List<String> parameterCars) {
+        assertThatExceptionOfType(IllegalArgumentException.class).isThrownBy(
+            () -> new Cars(parameterCars, 0)).withMessage("[ERROR] 자동차 개수는 적어도 1개 이상이어야 합니다.");
+    }
+
+    private static Stream<Arguments> Car들_움직임_확인() {
+        return Stream.of(
+            Arguments.of(4, Arrays.asList(new Car("a", 2), new Car("b", 4), new Car("C", 1))),
+            Arguments.of(5, Arrays.asList(new Car("a", 2), new Car("b", 4), new Car("C", 1))),
+            Arguments.of(6, Arrays.asList(new Car("a", 2), new Car("b", 4), new Car("C", 1))),
+            Arguments.of(7, Arrays.asList(new Car("a", 2), new Car("b", 4), new Car("C", 1))),
+            Arguments.of(8, Arrays.asList(new Car("a", 2), new Car("b", 4), new Car("C", 1))),
+            Arguments.of(9, Arrays.asList(new Car("a", 2), new Car("b", 4), new Car("C", 1))));
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    void Car들_전진_움직임_확인(int movableNumber, List<Car> parameterCars) {
+        Cars cars = new Cars(parameterCars);
+        List<Car> moveCars = new ArrayList<>();
+        for (Car car : parameterCars) {
+            moveCars.add(new Car(car.getNameValue(), car.getPositionValue() + 1));
+        }
+        assertThat(cars.driveCars(() -> movableNumber)).isEqualTo(moveCars);
```

Comment:
아래와 같이 cars 객체의 동등성 비교로 값을 검증하기 보다는 내부 값을 직접 꺼내서 비교해주세요! 😃
테스트만을 위해서 equals & hashCode를 정의하는 것은 불필요한 구현이라고 생각해요. 👀
```suggestion
        List<Car> actual = cars.driveCars(() -> movableNumber);
        
        assertThat(actual.get(0)).isEqualTo(xxxxxx);
        assertThat(actual.get(1)).isEqualTo(yyyyyy);
        ....
```

Reply:
넵! ㅎㅎ 직접 꺼내서 검증하는 형태로 구현해보겠습니다!